e2e/LongHaul/module/IotHub.cs

```diff
@@ -41,7 +42,8 @@ internal sealed class IotHub : IAsyncDisposable
         public IotHub(Logger logger, Parameters parameters)
         {
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
-            _moduleConnectionString = parameters.ConnectionString;
+            _moduleConnectionString = parameters.GatewayHostName == null ?
```

Comment:
Instead of messing with the connection string, can we just use the ClientOptions.GatewayHostname property?